Gems/ROS2/Code/Source/Spawner/ROS2SpawnPointComponent.h

```diff
@@ -6,7 +6,9 @@
  *
  */
 #pragma once
-
+#ifndef WITH_GAZEBO_MSGS
```

Comment:
Could we replace these repetitive asserts and messages with a single CMAKE check (message / fail) instead? Regardless, it is enough to place them in the header files.

Reply:
I can remove those ifdefs completely. From my own experience I prefer to have clear, early compilation errors when something is badly configured.

Reply:
Removed those pragmas.